api/src/lib/integrations/ariUtils.ts

```diff
@@ -241,47 +241,43 @@ export const handleIncomingARI = async (question: I.ARIQuestion): Promise<I.Hand
     const changes = detectChangesToARIPublication(mappedData, existingVersion);
 
     if (changes) {
-        // Data differs from what is in octopus, so attempt to re-version the publication.
-
-        // If the user (ARI department) changes, the create() call handles that.
-        // As such, we don't need to deal with it in the update() call.
-        const newVersion = await publicationVersionService.create(existingVersion, user);
-        // Mock coauthor workflow process of stating affiliations.
-        // Treat organisational account as an independent (unaffiliated) coauthor.
-        await coAuthorService.update(newVersion.coAuthors[0].id, { isIndependent: true });
-
-        const updatedVersion = await publicationVersionService.update(newVersion.id, {
+        // Data differs from what is in octopus, so update the publication.
+        // Unlike manually created publications, these just have 1 version that
+        // updates in-place so that we don't pollute datacite with lots of version DOIs.
+        let updatedVersion = await publicationVersionService.update(existingVersion.id, {
             ...(changes.title && { title: mappedData.title }),
             ...(changes.content && { content: mappedData.content }),
             ...(changes.keywords && { keywords: mappedData.keywords }),
-            ...(changes.topics && { topics: { set: mappedData.topicIds.map((topicId) => ({ id: topicId })) } })
+            ...(changes.topics && { topics: { set: mappedData.topicIds.map((topicId) => ({ id: topicId })) } }),
+            ...(changes.userId && { user: { connect: { id: mappedData.userId } } })
         });
-        const isReadyToPublish = await publicationVersionService.checkIsReadyToPublish(updatedVersion);
-
-        if (isReadyToPublish) {
-            await publicationVersionService.generateNewVersionDOI(updatedVersion, existingVersion);
-            const publishedNewVersion = await publicationVersionService.updateStatus(newVersion.id, 'LIVE');
-
-            if (publishedNewVersion) {
-                return {
-                    actionTaken: 'update',
-                    success: true,
-                    publicationVersion: publishedNewVersion
-                };
-            } else {
-                return {
-                    actionTaken: 'update',
-                    success: false,
-                    message: `Created a new version (id: ${newVersion.id}) of the existing publication for ARI question: ${question.questionId}, but failed to update its status to LIVE.`
-                };
+
+        // If user changed, update coAuthors.
+        if (changes.userId) {
+            // Delete old coAuthor.
+            // There should only be one coAuthor but since this is an array, loop through to make it clean.
+            await Promise.all(
+                existingVersion.coAuthors.map(async (coAuthor) => {
```

Comment:
You can just return the promise from `deleteCoAuthor` rather than awaiting that promise and wrapping the callback in it's own promise.

Reply:
```
await Promise.all(existingVersion.coAuthors.map(coAuthor => coAuthorService.deleteCoAuthor(coAuthor.id));
```